### ISSUE TITLE
Mobile VR Interface doesn't work in 4.X

### ISSUE THREAD

**Issue by AdriLorenz** [godotengine/godot#89755]
### Tested versions

Tested in Godot 4.2 and 4.3 dev5

### System information

Samsung Galaxy S23 Ultra, Godot in compatibily mode 

### Issue description

The interface that manages VR mobile games called 'Native mobile' is totally broken right now and it doesn't rotate the camera properly. It also has a weird flickering.

### Steps to reproduce

- Create a scene and enable the XR shaders.
- Add an XROrigin as a child of a Node3D and XRCamera inside of it.
- Add a script with the following code:

```
extends Node3D

var xr_interface: XRInterface

func _ready():
	var interface = XRServer.find_interface("Native mobile")
	if interface and interface.initialize():
		get_viewport().use_xr = true

```

- Add some floor or an environment to check the rotation
- Export to Android and check the rotation

### Minimal reproduction project (MRP)

[NativeMobileBug.zip](https://github.com/godotengine/godot/files/14702924/NativeMobileBug.zip)

**Comment by HectorVarel** [comment]
Do you resolve this problem? I have the same:(

**Comment by AdriLorenz** [comment]
> Do you resolve this problem? I have the same:(

Nothing. This seems to be out of the priorities of the Godot Team, sadly x.x

**Comment by bierig** [comment]
I can confirm this issue -- it blew a VR lab for me out of purpose. Please address this or get in contact with me so that I can help getting started.

**Comment by BastiaanOlij** [comment]
@akien-mga  just having a quick look, I'm using a Pixel 6 phone to test. I'm not getting any flickering issues but indeed the orientation code is far more broken than I previously thought so I'll investigate that further.

On the flickering issue, it would be good to know what GPU is on the phone. I only have Mali phones here to test. If the GPU is a Qualcomm (which googling suggests it does) it could be the same issue as we're experiencing on Quest in which case it would be good for @DarioSamo to take a look at it as the native mobile setup doesn't do anything special with rendering.

On a side note, I did notice that glow is completely broken when running on a mobile phone, while working properly on desktop. Thats a whole new issue...

**Comment by DarioSamo** [comment]
> On the flickering issue, it would be good to know what GPU is on the phone. I only have Mali phones here to test. If the GPU is a Qualcomm (which googling suggests it does) it could be the same issue as we're experiencing on Quest in which case it would be good for @DarioSamo to take a look at it as the native mobile setup doesn't do anything special with rendering.

We clarified this over DM already, but OP claims to be using the compatibility renderer, an area I haven't really worked on, so I'm not familiar with what could be affecting it in that regard.

**Comment by BastiaanOlij** [comment]
Yup, read that wrong.. I can't reproduce the flickering on compatibility either (yet), still doing some tests.

Did fix the orientation issue, that was dumb, I should have found that ages ago.

**Comment by bierig** [comment]
The orientation issue is the one that is most pressing imho as it prevents mobile VR altogether.

**Comment by BastiaanOlij** [comment]
> The orientation issue is the one that is most pressing imho as it prevents mobile VR altogether.

Yes, but that should be fixed by #91305, it would be good to confirm.

**Comment by akien-mga** [comment]
Assuming fixed by #91305.
